src/gui/SubWindow.cpp

```diff
@@ -26,42 +26,250 @@
+	m_windowTitle->setGraphicsEffect(m_shadow);
+}
+
+
+
+
+void SubWindow::paintEvent(QPaintEvent *)
+{
+	QPainter p(this);
+	QRect rect(0, 0, width(), m_titleBarHeight);
+	bool isActive = SubWindow::mdiArea()->activeSubWindow() == this;
+
+	p.fillRect(rect, isActive ? activeColor() : p.pen().brush());
+
+	// window border
+	p.setPen(borderColor());
+
+	// bottom, left, and right lines
+	p.drawLine(0, height() - 1, width(), height() - 1);
+	p.drawLine(0, m_titleBarHeight, 0, height() - 1);
+	p.drawLine(width() - 1, m_titleBarHeight, width() - 1, height() - 1);
+
+	//window icon
+	QPixmap winicon(widget()->windowIcon().pixmap(m_buttonSize));
+	p.drawPixmap(3, 3, m_buttonSize.width(), m_buttonSize.height(), winicon);
+}
+
+
+
+
+void SubWindow::elideText(QLabel *label, QString text)
+{
+	QFontMetrics metrix(label->font());
+	int width = label->width() - 2;
+	QString clippedText = metrix.elidedText(text, Qt::ElideRight, width);
+	label->setText(clippedText);
 }
 
+
+
+
 QRect SubWindow::getTrueNormalGeometry() const
 {
 	return m_trackedNormalGeom;
 }
 
+
+
+
+QBrush SubWindow::activeColor() const
+{
+	return m_activeColor;
+}
+
+
+
+
+QColor SubWindow::textShadowColor() const
+{
+	return m_textShadowColor;
+}
+
+
+
+
+QColor SubWindow::borderColor() const
+{
+	return m_borderColor;
+}
+
+
+
+
+void SubWindow::setActiveColor(const QBrush & b)
+{
+	m_activeColor = b;
+}
+
+
+
+
+void SubWindow::setTextShadowColor(const QColor & c)
+{
+	m_textShadowColor = c;
+}
+
+
+
+
+void SubWindow::setBorderColor(const QColor &c)
+{
+	m_borderColor = c;
+}
+
+
+
+
 void SubWindow::moveEvent(QMoveEvent * event)
 {
 	QMdiSubWindow::moveEvent(event);
 	// if the window was moved and ISN'T minimized/maximized/fullscreen,
 	//   then save the current position
-	if (!isMaximized() && !isMinimized() && !isFullScreen())
+	if(!isMaximized() && !isMinimized() && !isFullScreen())
 	{
 		m_trackedNormalGeom.moveTopLeft(event->pos());
 	}
 }
 
+
+
+
 void SubWindow::resizeEvent(QResizeEvent * event)
 {
+	/* button adjustments*/
+	m_minimizeBtn->hide();
+	m_maximizeBtn->hide();
+	m_restoreBtn->hide();
+
+	const int rightSpace = 3;
+	const int buttonGap = 1;
+	const int menuButtonSpace = 24;
+
+	QPoint rightButtonPos( width() - rightSpace - m_buttonSize.width() , 3 );
+	QPoint middleButtonPos( width() - rightSpace - ( 2 * m_buttonSize.width() ) - buttonGap, 3 );
+	QPoint leftButtonPos( width() - rightSpace - ( 3 * m_buttonSize.width() ) - ( 2 * buttonGap ), 3 );
+
+	//The buttonBarWidth relates on the count of button.
+	//We need it to calculate the width of window title label
+	int buttonBarWidth = rightSpace + m_buttonSize.width();
+
+	//set the buttons on their positions.
+	//the close button is ever needed and on the rightButtonPos
+	m_closeBtn->move(rightButtonPos);
+
+	//here we ask: is the Subwindow maximizable and/or minimizable
+	//then we set the buttons and show them if needed
+	if ( windowFlags() & Qt::WindowMaximizeButtonHint )
```

Comment:
I'd suggest to remove the space that comes after `if`. Generally, you need to get your spaces in order. Having space or no space are both valid options, but it should be consistent across the whole file.